utils/stm-to-jitx.stanza

```diff
@@ -42,31 +43,35 @@ public defn to-jitx-pin-properties (pin-properties:STMPinProperties) -> False :
     if not numeric-pads? and not named-pads? :
       fatal("Generated a micro-controller without pads.") 
     
+    ; We need to compile a list of reused pad to pin mappings.
+    val pad-mappings = HashTable<String|Int, Vector<String>>()
+    defn check-pad (pad:String|Int, pin:String):
+      if get?(pad-mappings, pad) is False : 
+        pad-mappings[pad] = Vector<String>()
+      add(pad-mappings[pad], pin)
+      length(pad-mappings[pad]) == 1
```

Comment:
You can set a default vector for a table. This lets you add a value without checking if the entry already exists.
```
    val pad-mappings = HashTable-init<String|Int, Vector<String>>(Vector<String>{})
    defn check-pad (pad:String|Int, pin:String) -> True|False :
      add(pad-mappings[pad], pin)
      length(pad-mappings[pad]) == 1
```

Reply:
Good trick, thanks!

Reply:
Using 

```
    val pad-mappings = HashTable<String|Int, Vector<String>>(Vector<String>())
```

Caused a lookup error, so I'm leaving the original for now.

Reply:
Sure, yours works too. In the future make sure to use curlies `...(Vector<String>{})`